vuu-ui/packages/vuu-utils/src/column-utils.ts

```diff
@@ -1210,3 +1214,20 @@ export const dataAndColumnUnchanged = (
   p.column === p1.column &&
   p.column.valueFormatter(p.row[p.columnMap[p.column.name]]) ===
     p1.column.valueFormatter(p1.row[p1.columnMap[p1.column.name]]);
+
+/**
+ * A memo compare function for cell renderers. Can be used to suppress
+ * render where column, row key  and data are all unchanged. Avoids render
+ * when row changes, where changes in row are unrelated to this cell.
+ * Suitable for editable cells. Including key in compare is not strictly
+ * necessary for rendering, but it is important in the event that user
+ * edits data - ensures we never have a stale key.
+ */
+export const dataColumnAndKeyUnchanged = (
+  p: TableCellRendererProps,
+  p1: TableCellRendererProps
+) =>
+  p.column === p1.column &&
+  p.row[KEY] === p1.row[KEY] &&
+  p.column.valueFormatter(p.row[p.columnMap[p.column.name]]) ===
+    p1.column.valueFormatter(p1.row[p1.columnMap[p1.column.name]]);
```

Comment:
Could've reused the above `dataAndColumnUnchanged` to avoid duplication. I know it's already merged, just leaving this here for future reference.

Reply:
Good point will tidy in one of my next PRs